hw3/exercise-03/src/pages/Houses.jsx

```diff
@@ -0,0 +1,162 @@
+    "rgba(212, 199, 199, 1)",
+    "rgba(72, 52, 199, 1)",
+  ];
+
+  // url for the Thrones API
+  const url = "https://thronesapi.com/api/v2/Characters";
+
+  // Collect houses as labels
+  const houses = [];
+
+  // Houses with their associated # of characters
+  const houseCount = [];
+
+  // Collect house's counts to insert into chart
+  const count = [];
+
+  // Fetch data from API asyncronously
+  const getData = async function fetchDataFromURL() {
+    try {
+      const response = await fetch(url);
+      console.log("Request successful", response);
+
+      // Now turn data into a json readable format
+      const data = await response.json();
+
+      // Initialize house & count values
+      getHouses(data);
+      getCount();
+    } catch (error) {
+      // Error from API fetch
+      console.error("Request failed", error);
+    }
+  };
+
+  const getHouses = function getHousesFromData(data) {
+    // Loop through data
+    data.forEach((data_item) => {
+      // Strip 'house' from family names
+      data_item.family = data_item.family.replace("House ", "");
+
+      // Address inconsistencies with Lannister family
+      if (data_item.family === "Lanister") {
+        data_item.family = "Lannister";
+      }
+
+      // Address inconsistencies with unknown/none/empty family inputs
+      if (
+        data_item.family === "Unkown" ||
+        data_item.family === "None" ||
+        data_item.family === ""
+      ) {
+        data_item.family = "Unknown";
+      }
+
+      // Address inconsistencies with Targaryen
+      if (data_item.family === "Targaryan") {
+        data_item.family = "Targaryen";
+      }
+
+      // Address inconsistencies with Lorath
+      if (data_item.family === "Lorathi") {
+        data_item.family = "Lorath";
+      }
+
+      // We will not be updating family value from here, save as constant
+      const family = data_item.family;
+
+      if (houses.includes(family)) {
+        // Family already exists in list of houses, add to count
+        const index = houseCount.map((e) => e.family).indexOf(family);
+        houseCount[index].count += 1;
+      } else {
+        // Family does not exists yet in list, add new entry & start new count
+        houses.push(family);
+        houseCount.push({ family: family, count: 1 });
+      }
+    });
+  };
+
+  // Function to get values of each house count to insert into chart
+  const getCount = function getCountFromCountArray() {
+    houseCount.forEach((count_item) => {
+      count.push(parseInt(count_item.count));
+    });
+  };
+
+  // Initialize everything
+  getData();
+
+  const data = {
+    labels: houses,
+    datasets: [
+      {
+        label: "Houses",
+        data: count,
+        backgroundColor: backgroundColors,
+        borderColor: borderColors,
+        borderWidth: 1,
+      },
+    ],
+  };
+
+  return (
+    <>
+      <h1>Houses</h1>
+      <Doughnut data={data} />
+    </>
+  );
+};
+export default Houses;
```

Comment:
I ran your code and it looked great and functioning, however it seems like (as you mentioned in your PR request) you didn't include grouping the data.